src/net.cpp

```diff
@@ -1999,11 +1999,30 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             if (nTries > 100)
                 break;
 
-            CAddrInfo addr = addrman.SelectTriedCollision();
+            CAddrInfo addr;
 
-            // SelectTriedCollision returns an invalid address if it is empty.
-            if (!fFeeler || !addr.IsValid()) {
-                addr = addrman.Select(fFeeler);
```

Comment:
Very happy to see this rewritten. The logic where we always called `SelectTriedCollision` and then throw out the result if `!fFeeler` was very confusing (and inefficient...).